src/index.js

```diff
@@ -85,41 +103,44 @@ export const notificationsChannelURLFromHookURL = hookURL => {
 const handleIncoming = event => {
   const data = JSON.parse(event.data);
 
-  // init response
-  // identified by presence of user_id_cookie
-  if (data.user_id_cookie) {
+  const isInitResponse = data.user_id_cookie;
+  const isErrorResponse = data.error;
+
+  if (isInitResponse) {
     if (!store.userId || store.userId !== data.user_id_cookie) {
       store.userId = data.user_id_cookie;
       Cookies.set(COOKIE_NAME, store.userId);
 
+      cleanSessionHistory();
+
       if (store.promises.getUserId) {
         store.promises.getUserId.resolve(store.userId);
         store.promises.getUserId = undefined;
       }
     }
 
-    // if messages were queued then they need to be appended after history
     if (store.promises.init.resolve) {
-      const history = cleanHistory(data.history);
+      const history = cleanHistory(getSessionHistory());
 
       store.messageQueue.forEach(message => {
         if (message !== CONVERSATION_STARTER) {
-          history.push({ made_by: 'user', content: message });
+          pushSessionHistoryItem({ made_by: 'user', content: message });
```

Comment:
you don't have to push it to history again right?
You retrieve it (line 103) and give it to widget-core consumer

Reply:
When widget-core consumer sends a message before connection is established, the message is pushed to a queue. The consumer will add the messages to their UI.
Once connection is established, new history is retrieved and given to the widget-core consumer. The consumer will reset their UI with messages from history and hence we need to add the messages from the queue to this history. We also need to add it to local storage history for the next time, this you are doing inside `send`.